client/sites/hooks/test/use-transfer-with-software-start-mutation.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import nock from 'nock';
+import React from 'react';
+import { useRequestTransferWithSoftware } from '../use-transfer-with-software-start-mutation';
+
+const replyErrorWithEnvelope =
+	( status: number, defaultBody: Record< string, string | number > = {} ) =>
+	( body = {} ) =>
+	() => [ 200, { code: status, body: { ...defaultBody, ...body } } ];
+const SITE_ID = 123;
+const API_SETTINGS = { migration_source_site_domain: 'example.com' };
+
+const Wrapper =
+	( queryClient: QueryClient ) =>
+	( { children } ) => (
+		<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+	);
+
+const render = ( options = { retry: 0 } ) => {
+	const queryClient = new QueryClient();
+
+	const renderResult = renderHook(
+		() =>
+			useRequestTransferWithSoftware(
+				{
+					siteId: SITE_ID,
+					apiSettings: API_SETTINGS,
+					plugins: [ 'plugin-1', 'install' ],
+					themes: [ 'theme-1', 'activate' ],
+				},
+				options
+			),
+		{
+			wrapper: Wrapper( queryClient ),
+		}
+	);
+
+	return {
+		...renderResult,
+		queryClient,
+	};
+};
+
+describe( 'useRequestTransferWithSoftware', () => {
+	beforeAll( () => {
+		nock.disableNetConnect();
+	} );
+
+	beforeEach( () => nock.cleanAll() );
+
+	it( 'should successfully request transfer with software and return the transfer_with_software_id', async () => {
```

Comment:
Hey @sixhours,
After reading your comment here(175730-ghe-Automattic/wpcom#discussion_r167753) I realized that you used the `transfer_with_software_id` key. We won't have the `transfer_with_software_id` because we decided to move on without creating the table and taking advantage of the transfer meta table. This means that we should use the `atomic_transfer_id` to identify the status of the migration. We should refer to paYKcK-61R-p2. I will mark the other post as deprecated. Sorry for not making it clear!